src/web/js-interop.md

```diff
@@ -22,3 +22,124 @@ For help using the `js` package, see the following:
 [js-api]: {{site.pub-api}}/js
 [sass]: {{site.pub-pkg}}/sass
 
+## Next-generation JS interop preview
+
+{{site.alert.note}}
+  This interop feature is **experimental**,
+  and [in active development](https://github.com/dart-lang/sdk/issues/35084).
+{{site.alert.end}}
+
+Dart's JS interop story is currently evolving. 
+Many of the features that enable future JS interop
+are ready to experiment with as of Dart version 2.19.
+These features support the existing production
+and development web compilers, as well as Dart's
+in-progress Wasm compiler ([`dart2wasm`][]).
+
+For a glimpse into the next generation of JS interop,
+you can refactor your code to conform to the new
+syntax and semantics now. Doing so will
+likely not prevent the need to refactor again once
+[Dart 3][] lands, as the features are still in development. 
+However, the features available for preview are much
+closer to future JS interop than any pattern supported today,
+so there are a few reasons to try them out now:
+* Potentially ease transition of existing JS
+interop code once migration becomes necessary.
+* Existing JS interop developers eager to prepare their code for Dart 3.
```

Comment:
Technically we don't plan to break anything by Dart 3. Consider combining this and the bullet below to present it as all the future exicting things happening? For example:

* Existing JS interop developers eager to experiment with the latest features in JS interop or with `dart2wasm` when it becomes available.